pkg/controller/extension/actuator.go

```diff
@@ -123,6 +123,11 @@ func (a *actuator) Migrate(_ context.Context, _ logr.Logger, _ *extensionsv1alph
 	return nil
 }
 
+// ForceDelete the Extension resource.
+func (a *actuator) ForceDelete(ctx context.Context, log logr.Logger, ext *extensionsv1alpha1.Extension) error {
+	return a.Delete(ctx, log, ext)
```

Comment:
I was wondering whether the following will work. The extension deploys a StatefulSet with PVC (via ManagedResource) for every configured registry-cache in the Shoot. Respectively, on deletion the extension deletes the corresponding ManagedResource which deletes the StatefulSet and the PVC. 
When the Shoot has invalid cloud provider credentials (or credentials with insufficient privileges) how can the PVC get deleted then? I assume in that case the force deletion won't work and the extension will stuck waiting for the PVC to be deleted? Did you test the scenario?

Reply:
@ialidzhikov I have updated the PR. In case of force delete it only deletes MR and doesn't wait to clean registry cache configuration with the assumption that user would clean up backing infra resources.

Also opened a [PR](https://github.com/gardener/gardener/pull/8710) to add some examples of what to do in case of force delete - https://github.com/gardener/gardener/blob/3513ee00afe7f1cacd7758530157c7d2225795ca/docs/extensions/force-deletion.md#some-general-force-delete-implementation-examples

Reply:
Thanks!

>In case of force delete it only deletes MR and doesn't wait to clean registry cache configuration with the assumption that user would clean up backing infra resources.

You share that the registry-cache extension should only delete the MR and should not wait for the MR deletion.
But in the current revision of the PR 
```go
	if err := component.OpDestroyAndWait(registryCaches).Destroy(ctx); err != nil {
		return fmt.Errorf("failed to destroy the registry caches component: %w", err)
	}
```

is used.

So I assume the intension was to use `component.Destroy(ctx)` (without waiting)?

Reply:
`component.OpDestroyAndWait(registryCaches).Destroy(ctx);` will also not wait for MR to gone. Is that the case?
To wait for MR to be gone, we have to call the `WaitCleanup`.

Reply:
It should wait it (call `WaitCleanup` under the hood).

https://github.com/gardener/gardener-extension-registry-cache/blob/00056ca05aba1b70e4e5ce902622324c8f24ed53/vendor/github.com/gardener/gardener/pkg/component/deploy.go#L38-L44

https://github.com/gardener/gardener-extension-registry-cache/blob/00056ca05aba1b70e4e5ce902622324c8f24ed53/vendor/github.com/gardener/gardener/pkg/component/deploy.go#L88-L106

Reply:
Ahh, Thanks. Will update the PR.